tests/unit/states/test_win_task.py

```diff
@@ -0,0 +1,507 @@
+                               'ac_only': True},
+                'settings': {'force_stop': True,
+                             'allow_demand_start': True,
+                             'delete_after': False,
+                             'multiple_instances': 'No New Instance',
+                             'execution_time_limit': '3 days',
+                             'stop_if_on_batteries': True,
+                             'restart_interval': False,
+                             'wake_to_run': False}}
+
+        with patch.dict(win_task.__salt__, {'task.list_tasks': MagicMock(side_effect=[['salt']] * 2),
+                                            'task.info': MagicMock(side_effect=[info])}), \
+                patch.dict(win_task.__opts__, {"test": True}), \
+                patch.dict(win_task.__grains__, {'osversion': '7.1'}):
+
+            ret = win_task.present(name='salt', location='', force=True, **kwargs)
+            self.assertEqual(ret['result'], True)
+
+    def test_OnSessionChange(self):
+        kwargs = {'action_type': 'Execute',
+                  'cmd': 'del /Q /S C:\\\\Temp',
+                  'trigger_type': 'OnSessionChange',
+                  'start_date': '2019-05-14',
+                  'start_time': '01:00 pm',
+                  'state_change': 'SessionUnlock'}
+
+        info = {'actions': [{'cmd': 'del /Q /S C:\\\\Temp',
+                             'action_type': 'Execute'}],
+                'settings': {'delete_after': False,
+                             'execution_time_limit': '3 days',
+                             'wake_to_run': False,
+                             'force_stop': True,
+                             'multiple_instances': 'No New Instance',
+                             'stop_if_on_batteries': True,
+                             'restart_interval': False,
+                             'allow_demand_start': True},
+                'triggers': [{'trigger_type': 'OnSessionChange',
+                              'execution_time_limit': '3 days',
+                              'delay': False,
+                              'enabled': True,
+                              'start_date': '2019-05-14',
+                              'start_time': '13:00:00'}],
+                'conditions': {'run_if_idle': False,
+                               'ac_only': True,
+                               'run_if_network': False,
+                               'start_when_available': False}}
+
+        with patch.dict(win_task.__salt__, {'task.list_tasks': MagicMock(side_effect=[['salt']] * 2),
+                                            'task.info': MagicMock(side_effect=[info])}), \
+             patch.dict(win_task.__opts__, {"test": True}), \
+             patch.dict(win_task.__grains__, {'osversion': '7.1'}):
+            ret = win_task.present(name='salt', location='', force=True, **kwargs)
+
+            self.assertEqual(ret['result'], True)
```

Comment:
These test functions should not be `camelCased`, use the Zen of python.
Instead of `def test_OnSessionChange(self):` do `def test_on_session_change(self):`.